tools/varint/build.gradle.kts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+plugins { id("polaris-server") }
+
+dependencies {
+  implementation(libs.guava)
+
+  testFixturesApi(libs.assertj.core)
+}
+
+description = "Provides variable length integer encoding"
```

Comment:
Do we need to create a new module for just one class? I'm not against having multiple modules, but I'd like to understand the benefits.

My concerns are:
1. Increased Complexity – If we continue this approach, we'll end up with significantly more modules to maintain, which could add overhead.
2. Impact on Downstream Users – More modules mean more dependencies, potentially making it harder for downstream users who will have to manage a lot of Polaris jars, other than just a few of them.

Would love to hear the rationale behind this approach!

Reply:
The goal here is to have one dependency that does not have unnecessary other dependencies, kryo for example comes with a bunch of other dependencies (so "even more jars").

The opposite, not sure if you're proposing that, is to have monoliths. The disadvantage is what we currently have in the code base - a bunch of unrelated things that depend on each other.

I don't see an impact on downstream users because:
a) versions are consistently managed via the bom
b) dependencies are transitive, not manual - unless you're using the ant-ique build tool, you're fine.

Reply:
My concern here is more about the maintenance overhead rather than the impact on downstream users. Of course, if we write perfect bug-free code downstream users are not impacted. But if we can avoid a new module, and indeed even new code to maintain, I think we should explore that option.

Reply:
Please note that this one is used by a bunch of modules in #1189. This piece is "tailor made" for those use cases. Kryo's serialization is different, and brings full-blown object mapping + alternative logging - quite too much.

`polaris-core` already has way too may things - depends on Iceberg and public API modules - things on which persistence work should really not depend on.

Reply:
Kryo is one example, but I'm not partial to it over any other dependency we might pull in for serialization. I do not think we really want to be in the business of maintaining some custom serialization module when there is so much prior art.

On the topic of modules, if this is tailor-made for one persistence implementation, can we start by putting it there?

Reply:
I'm generally open to move it closer there, but I'd prefer to merge this one and make progress with the overall large #1189 work and spill out smaller chunks from that PR. This is not anything that's user visible.

Reply:
> polaris-core already has way too may things - depends on Iceberg and public API modules - things on which persistence work should really not depend on.

The persistence work will depend on `polaris-core` anyway, esp. the classes under the package `org.apache.polaris.core.persistence`.

Reply:
> The persistence work will depend on `polaris-core` anyway, esp. the classes under the package `org.apache.polaris.core.persistence`.

That's not entirely true. The parts that deal with low-level things do _not_ depend on that one.

Again, I'm not opposed to change things later, but for now I'd like to stick with this one to unblock the whole #1189 effort.

Reply:
This code is fairly simple and very unit-testable. I do not see a reason to pull in kryo (with it's unrelated baggage). A local impl. fulfils Polaris persistence concerns well enough, from my POV.

Reply:
If the Polaris concerns are specific to persistence -- and it sounds like one (future) persistence implementation -- can we put the code in the appropriate place for that?

Reply:
@eric-maynard : by "appropriate" do you mean somewhere under `/persistence`? That would work for me.

Reply:
If only one module needs this, I think this code can probably live in that module for now

Reply:
I see... but as discussed in the persistence community meeting, I thought people wanted the NoSQL code to arrive in smaller, easier to review chunks :thinking:

Reply:
small PRs does not mean many small modules

Reply:
I won't object to folding `Varint` into one of the modules that come later. Perhaps we could have the module skeleton in this PR and add other classes later... Does it work for you, @snazy ?